packages/core/src/browser/quick-open/prefix-quick-open-service.ts

```diff
@@ -154,7 +154,9 @@ export class PrefixQuickOpenService {
     protected currentHandler: QuickOpenHandler | undefined;
 
     protected async setCurrentHandler(prefix: string, handler: QuickOpenHandler | undefined): Promise<void> {
-        this.toDisposeCurrent.dispose();
+        if (handler !== this.currentHandler) {
+            this.toDisposeCurrent.dispose();
```

Comment:
@azatsarynnyy good catch! We should move next 2 lines in the if as well, otherwise, it adds more and more disposables

Reply:
@akosyakov I've fixed disposing of a handler multiple times.